sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala

```diff
@@ -1610,6 +1610,17 @@ class AstBuilder extends SqlBaseBaseVisitor[AnyRef] with SQLConfHelper with Logg
     cast
   }
 
+  /**
+   * Create a [[TryCast]] expression.
+   */
+  override def visitTryCast(ctx: TryCastContext): Expression = withOrigin(ctx) {
```

Comment:
`visitCast` and `visitTryCast` are similar between each other, so how about merging their definition in `SqlBase.sql`?
```
    | cast=(CAST | TRY_CAST) '(' expression AS dataType ')'                                                      #cast
```